### ISSUE TITLE
ORBAT Creator errors out when inheriting classes that do not contain Eventhandlers.CBA_Extended_EventHandlers_base

### ISSUE THREAD

**Issue by Yedidya-SPHG** [ALiVEOS/ALiVE.OS#329]
Hello everyone,

I have created a custom faction based on the RHSAFRF mod, since RHS does not group properly and ALiVE complains about not having anything in the groups. Now the issue I have is, after I created a full faction etc, that my ArmA 3 won't even start and throws the following Error:

File russia_faction_sphg\autogen.hpp, line 3871:
/CfgVehicles/O_russiafactionsphg_Rifleman_01.EventHandlers:
Undefined base class 'EventHandlers'

Here you can find the logfile and the .pbo 

Logfile: https://www.dropbox.com/s/h568flsbr0hbxsy/arma3_x64_2017-04-19_12-58-29.rpt?dl=0
PBO: https://www.dropbox.com/s/h1x07flectayxkw/russia_faction_sphg.pbo?dl=0

Thanks a lot in advance!

Cheers,
Yedi

**Comment by SpyderBlack723** [comment]
Please make sure you are using the latest version of ALIVE to export the factions.

**Comment by Yedidya-SPHG** [comment]
Any way I can utilize the already done Faction, because invested about 3h, and don't want to do everything again from ground :(

**Comment by Yedidya-SPHG** [comment]
PS: If I can believe the changelog file in my ALiVE folder, I am using the latest version

**Comment by SpyderBlack723** [comment]
Please post your faction's autogen.hpp file output please, thanks.

**Comment by Yedidya-SPHG** [comment]
To copy paste it here, it is too long, hence I put a pastebin link here https://pastebin.com/WMrW7fa7

**Comment by Yedidya-SPHG** [comment]
Alright, I got it fixed on my end and also for issue #331 

The error was in the, I assume, alias conversion for the rhs units.

e.g. this line   class O_russiafactionsphg_Rifleman_01 : rhs_msv_emr_rifleman {

needs to be this way   class O_russiafactionsphg_Rifleman_01 : rhs_msv_emr_rifleman_OCimport_02 {

So by all the lines like that _OCimport_02 was missing

After I added it manually, it worked like a charm. But now I have another issue... whenever I want to start the ALiVE mission locally, it tells me in the beginning

There are no groups for OPCOM faction(s) ["russia_faction_sphg"]! Please check you chose the correct faction(s), and that factions have groups defined in the ArmA 3 default categories infantry, motorized, mechanized, armored, air, sea!"

As you can see above, I do have groups for a new faction, it shows it also properly in the 3den editor, but in Zeus it won't show them. The only thing I have not defined are sea groups. Any ideas?

**Comment by Yedidya-SPHG** [comment]
It shows it up in Zeus now, but ALiVE is still bitching :(

**Comment by coachwn** [comment]
Confirms it works for my issue as well.

**Comment by SpyderBlack723** [comment]
> As you can see above, I do have groups for a new faction, it shows it also properly in the 3den editor, but in Zeus it won't show them. The only thing I have not defined are sea groups. Any ideas?

Make sure you have at least one placement module that is spawning units for that faction, and not just a freestanding commander module.

**Comment by Yedidya-SPHG** [comment]
I do have all the modules placed which are needed and configured. If you want, I can take a screenshot of it and how they are connected to each other. When I go into a city, the CQB module places single units, but it will never spawn any groups, because it does not recognize the group unfortunately.

**Comment by SpyderBlack723** [comment]
Can you share your mission and a list of required mods if possible. At this point it could be a different issue.

**Comment by Yedidya-SPHG** [comment]
Hello SpyderBlack723,

I have uploaded my factions pbo file and mission file to google drive

https://drive.google.com/open?id=0B2RFBsPw6gyGMWlOTnczWjFxV2s

https://drive.google.com/open?id=0B2RFBsPw6gyGV2NMY0FLb1ozWUE

![mods](https://cloud.githubusercontent.com/assets/27768411/25248382/60157e08-260e-11e7-9f44-b684a0329f87.png)

**Comment by Yedidya-SPHG** [comment]
Seems like I got it to work ... I removed all the ALiVE modules again, after I tested mod by mod for incompatibility... placed the modules again, put the settings like before. Now it places groups and all that stuff. So I guess this can be closed. The original issue and its solution is written above, you might recheck your code on that one, why it did not add the _OCimport_02 & _OCimport_01 to almost all of the rhs units. Cheers for the fast responses.
